pkg/agent/config.go

```diff
@@ -57,6 +68,354 @@ type VenafiCloudConfig struct {
+	)
+	c.PersistentFlags().StringVarP(
+		&cfg.ClientID,
+		"client-id",
+		"",
+		"",
+		"Venafi Cloud Service Account client ID. If you use this flag you don't need to use --venafi-cloud as it will assume you are authenticating against Venafi Cloud. Using this removes the need to use a credentials file with Venafi Cloud mode.",
+	)
+	c.PersistentFlags().StringVarP(
+		&cfg.PrivateKeyPath,
+		"private-key-path",
+		"",
+		"/etc/venafi/agent/key/privatekey.pem",
+		"Venafi Cloud Service Account private key path.",
+	)
+	c.PersistentFlags().BoolVarP(
+		&cfg.OneShot,
+		"one-shot",
+		"",
+		false,
+		"Runs agent a single time if true, or continously if false",
+	)
+	c.PersistentFlags().StringVarP(
+		&cfg.OutputPath,
+		"output-path",
+		"",
+		"",
+		"Output file path, if used, it will write data to a local file instead of uploading to the preflight server",
+	)
+	c.PersistentFlags().StringVarP(
+		&cfg.InputPath,
+		"input-path",
+		"",
+		"",
+		"Input file path, if used, it will read data from a local file instead of gathering data from clusters",
+	)
+	c.PersistentFlags().DurationVarP(
+		&cfg.BackoffMaxTime,
+		"backoff-max-time",
+		"",
+		10*time.Minute,
+		"Max time for retrying failed data gatherers (given as XhYmZs).",
+	)
+	c.PersistentFlags().BoolVarP(
+		&cfg.StrictMode,
+		"strict",
+		"",
+		false,
+		"Runs agent in strict mode. No retry attempts will be made for a missing data gatherer's data.",
+	)
+	c.PersistentFlags().StringVar(
+		&cfg.APIToken,
+		"api-token",
+		os.Getenv("API_TOKEN"),
+		"Token used for authentication when API tokens are in use on the backend",
+	)
+	c.PersistentFlags().StringVar(
+		&cfg.VenConnName,
+		"venafi-connection",
+		"",
+		"Name of the VenafiConnection to be used. Using this flag will enable the VenafiConnection mode.",
+	)
+	c.PersistentFlags().StringVar(
+		&cfg.VenConnNS,
+		"venafi-connection-namespace",
+		"",
+		"Namespace of the VenafiConnection to be used. It is only useful when the VenafiConnection isn't in the same namespace as the agent. The field `allowReferencesFrom` must be present on the cross-namespace VenafiConnection for the agent to use it.",
+	)
+	c.PersistentFlags().StringVar(
+		&cfg.InstallNS,
+		"install-namespace",
+		"",
+		"Namespace in which the agent is running. Only needed when running the agent outside of Kubernetes. Used for testing purposes.",
+	)
+	c.PersistentFlags().BoolVarP(
+		&cfg.Profiling,
+		"enable-pprof",
+		"",
+		false,
+		"Enables the pprof profiling server on the agent (port: 6060).",
+	)
+	c.PersistentFlags().BoolVarP(
+		&cfg.Prometheus,
+		"enable-metrics",
+		"",
+		false,
+		"Enables Prometheus metrics server on the agent (port: 8081).",
+	)
+}
+
+// getConfiguration combines the input configuration with the flags passed to
+// the agent and returns the final configuration as well as the Venafi client to
+// be used to upload data.
+func getConfiguration(log *log.Logger, cfg Config, flags AgentCmdFlags) (Config, client.Client, error) {
```

Comment:
**Self-review:** I moved `getConfiguration` over from `run.go` as-is without any modification. This is because I think this belongs to `config.go`, not `run.go`. But let me know if it makes things too difficult to review.